### PR TITLE
EVEREST-356 Fixed pipeline 

### PR DESCRIPTION
**Pull request by gen1us2k**

### CODE REVIEW

File: controllers/databasecluster_controller.go [hunk 1]
```diff
@@ -2828,7 +2828,10 @@ func (r *DatabaseClusterReconciler) defaultPXCSpec() *pxcv1.PerconaXtraDBCluster
 					MaxUnavailable: &maxUnavailable,
 				},
 				Resources: corev1.ResourceRequirements{
-					Limits: corev1.ResourceList{},
+					Limits: corev1.ResourceList{
+						corev1.ResourceMemory: resource.MustParse("1G"),
```
Comment (gen1us2k) [inline controllers/databasecluster_controller.go:2832]: Sometimes it fails due to panic: cannot assign values to the nil map

**Review by oksana-grishchenko — COMMENTED** [review]
Great job, Andrew! I like the new structure of the tests and the idea of running them separately. Although I don't see a value for the CI user to see the steps like "Scale up" and "Scale down", I believe there is no easy way to organize the jobs differently if we want to run e2e tests in two different jobs.  

I am concerned that the features tests step exceeds the deadline and got skipped, is it something that is going to be fixed in the next PR?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -107,44 +127,144 @@ jobs:
...
           CLUSTER_NAME: "dbaas-ci-20221005"
           NODEPOOL_NAME: "terraform-node-dbaas-ci-20221005"
 
+      - name: Clean up resources and CRDs
+        if: always()
+        run: |
+          export KUBECONFIG="${HOME}/.kube/config"
+          mkdir -p "${KUBECONFIG%/*}"
+          echo "$BASE64_KUBECONFIG_DBAAS"|base64 -d > "${KUBECONFIG}"
+          kubectl delete db --all-namespaces --all
```
[the remark this answers is not in the document]
Comment (gen1us2k) [reply .github/workflows/ci.yml:144]: We should

**Comment by gen1us2k** [comment]
@oksana-grishchenko 

>Great job, Andrew! I like the new structure of the tests and the idea of running them separately. Although I don't see a value for the CI user to see the steps like "Scale up" and "Scale down", I believe there is no easy way to organize the jobs differently if we want to run e2e tests in two different jobs.

Scale up and scale down were moved to the separate job to ensure that EKS cluster scaled up before running tests. Right now it does not guarantee parallel execution in different PRs and there might be some cases when it conflicts between different runs and tests may run against scaled-down cluster.

The main idea is to scale up the cluster, run eks tests in parallel, and scale down when the execution is finished. I'll try to improve it a bit and make it more robust

> I am concerned that the features tests step exceeds the deadline and got skipped, is it something that is going to be fixed in the next PR?

Ah. I tested the restart of the operation. It runs without a timeout and I manually cancelled it. It's not an issue and it does not fail due timeouts

File: .github/workflows/ci.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (gen1us2k) [reply .github/workflows/ci.yml:144]: or we shouldn't. idk yet

**Review by oksana-grishchenko — APPROVED** [review]

**Review by michal-kralik — APPROVED** [review]

**Pull request merged**